### PR TITLE
Allow custom configuration for debugger diagnostics proxy

### PR DESCRIPTION
**Pull request by cimi**
### What does this PR do?

This PR allows custom configuration of the Live Debugger / Dynamic Instrumentation diagnostics proxy endpoint in the trace agent, in the same way we can configure the debugger proxy endpoint which forwards dynamic logs.

### Motivation

In environments where we publish data to multiple datacenters, the debugger diagnostics endpoint uses the default api key which is not necessarily the same one as for the default intake URL.

A possible solution would have been to reuse the API key set for the debugger proxy endpoint, since diagnostics and dynamic logs should always go to the same datacenter. For completeness, we allow the same configuration settings on the diagnostics endpoint like we do for the debugger proxy and the symdb proxy.

We do not dual / multi-publish debugger data at the moment, everything goes to a single datacenter (which is expected to be the datacenter the service is running in).

### Possible Drawbacks / Trade-offs

N/A this is adding new functionality. This functionality should only be used on the Datadog backend.

### Describe how to test/QA your changes

Added new unit tests, they all pass.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by ojung — APPROVED** [review]
Thanks, looking good from Dynamic Instrumentation / debugger perspective.

**Review by ogaca-dd — APPROVED** [review]

**Review by ajgajg1134 — CHANGES_REQUESTED** [review]
Overall looks good for trace-agent. Is there any documentation that needs to be updated here? Is this configuration something that customers will be using? If so can you add a changelog for this so that it's easy to determine from the release notes which version of the agent added support for these new variables?

**Comment by cimi** [comment]
> Overall looks good for trace-agent. Is there any documentation that needs to be updated here? Is this configuration something that customers will be using? If so can you add a changelog for this so that it's easy to determine from the release notes which version of the agent added support for these new variables?

Thanks for reviewing! AFAIK these configurations are not intended / used by customers. We use them internally when publishing data to multiple DCs.

Here's a similar PR that didn't add any other docs except the release note: https://github.com/DataDog/datadog-agent/pull/16409 

I'll add a release note here too :+1:

**Review by ajgajg1134 — APPROVED** [review]

File: releasenotes/notes/dynamic-instrumentation-diagnostics-dual-publishing-c23591af95b397d0.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+enhancements:
+  - |
+    Add DD_APM_DEBUGGER_DIAGNOSTICS_DD_URL, DD_APM_DEBUGGER_DIAGNOSTICS_API_KEY
```
Comment (maycmlee) [inline releasenotes/notes/dynamic-instrumentation-diagnostics-dual-publishing-c23591af95b397d0.yaml:4]:
```suggestion
    Add DD_APM_DEBUGGER_DIAGNOSTICS_DD_URL, DD_APM_DEBUGGER_DIAGNOSTICS_API_KEY,
```

**Review by maycmlee — APPROVED** [review]
Small nit, but approving.

**Pull request merged**